spanner/api/Spanner.Samples/WriteDatatypesDataAsync.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START spanner_insert_datatypes_data]
+
+using Google.Cloud.Spanner.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+public class WriteDatatypesDataAsyncSample
+{
+    public async Task WriteDatatypesDataAsync(string projectId, string instanceId, string databaseId)
+    {
+        string connectionString = $"Data Source=projects/{projectId}/instances/{instanceId}/databases/{databaseId}";
+
+        byte[] exampleBytes1 = Encoding.UTF8.GetBytes("Hello World 1");
+        byte[] exampleBytes2 = Encoding.UTF8.GetBytes("Hello World 2");
+        byte[] exampleBytes3 = Encoding.UTF8.GetBytes("Hello World 3");
+
+        var availableDates1 = new List<DateTime>();
+        availableDates1.InsertRange(0, new DateTime[] {
+                DateTime.Parse("2020-12-01"),
+                DateTime.Parse("2020-12-02"),
+                DateTime.Parse("2020-12-03")});
+
+        var availableDates2 = new List<DateTime>();
+        availableDates2.InsertRange(0, new DateTime[] {
+                DateTime.Parse("2020-11-01"),
+                DateTime.Parse("2020-11-05"),
+                DateTime.Parse("2020-11-15")});
+
+        var availableDates3 = new List<DateTime>();
+        availableDates3.InsertRange(0, new DateTime[] {
+                DateTime.Parse("2020-10-01"),
+                DateTime.Parse("2020-10-07")});
+        List<Venue> venues = new List<Venue> {
```

Comment:
I really think we should receive a IList<Venue> as parameter and insert that. And then, there's really nothing particular about this sample that would distinguish it from the InsertDataAsync sample, the library accepts the usual .NET types and "converts" them to the Spanner types. I'm fine if we need to show just that, i.e. there's nothing special to be done, but then I would remove all the data parsing and byte array initialization code. Users should know how to work with dates and byte arrays.
@skuruppu thoughts welcome.

Reply:
Hmmm I can't see this sample being used anywhere, but it is consistently implemented this way in other languages as well:
- [Ruby](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/8ce9a8db19d5f77dfca263d7064947e51160a901/spanner/spanner_samples.rb#L1068)
- [Golang](https://github.com/GoogleCloudPlatform/golang-samples/blob/62070bfa82e1a23e87ee84d563a43deb4b386d7c/spanner/spanner_snippets/spanner/spanner_insert_datatypes_data.go#L17)

We can talk about these offline and maybe include the docs team to clarify if we should delete them or not.

Reply:
Waiting for the docs team to confirm.